Botches the removal of proc_holders from tg after 2 years of it not being done

Szyszkrzyneczka opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

> Please see https://github.com/orgs/BeeStation/projects/23 for a status on testing.

## About The Pull Request

(mainly)Ports:

- https://github.com/tgstation/tgstation/pull/67083
- fixes #9874 


From testmerge:
- fixes #12054 
- fixes #12055

Also:

- https://github.com/tgstation/tgstation/pull/64124
- https://github.com/tgstation/tgstation/pull/65180
- https://github.com/tgstation/tgstation/pull/66604
- https://github.com/tgstation/tgstation/pull/76616

Mousepointer fix copy pasted from
- https://github.com/tgstation/tgstation/pull/57084
- https://github.com/tgstation/tgstation/pull/65589


This is now a ports proc_holder removal, action button refactor and antimagic refactor

## Why It's Good For The Game

Mostly for the refactor of spell code making it easier and possible to make crazy spells and also to make this codebase more closer to modernity
Also this doesn't affect the players 

## Testing Photographs and Procedure
Once this actually works I will put some screenshots here. Until then please direct my attention towards bits and bops of code that I might've borked up while porting this. Until then this will be a draft.

There is still a ton of work before this can even be considered for merging
<details>

Random ass actions (red ones are unavailable because i got no wizard robe)

![action1](https://github.com/user-attachments/assets/e95d357f-00cb-4d20-8929-4e6957c9a9f3)

Action palette thing (very cool)

![action2](https://github.com/user-attachments/assets/bf046ee1-6b74-4c76-bb48-53f85f53325d)


Cooldown test

![action3](https://github.com/user-attachments/assets/2df87355-0401-4428-a703-6ad48c8c4fe3)


Action palette test

![action4](https://github.com/user-attachments/assets/adcd64ad-c7d7-4772-95f2-f41984ab6620)

<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl:
code: Removes proc_holders in favour of using /datum/action and expanding on it some more
refactor: refactors how spell code works making it badass and stuff
tweak: Many spells changed to conform with the new code order, included is telepathy that now works by providing the user with a list of targets instead of clicking on one
tweak: Spell cooldowns are very likely to be slightly different (Actually readable) Old cooldown code is an unreadable mess
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Szyszkrzyneczka committed:
Taken all the files from the base pr

Second commit will remove all the errors
In the meantime pretty please tell me if you people want anything else to go with this pr

Tsar-Salat commented on code/__DEFINES/span.dm:
```diff
@@ -0,0 +1,179 @@
+// Sorted alphabetically
+#define span_abductor(str) ("<span class='abductor'>" + str + "</span>")
```
inb4 maint comment, but this specific code got flak in the past and ultimately closed.

https://github.com/BeeStation/BeeStation-Hornet/pull/4971

I think the reasoning for the denial was stupid, considering most other modern codebases use it and have made peace with it, yet contribs here still have the write out the whole damn thing every time. (Which is only furthering our code debt imo)

Just something to keep in mind.

Szyszkrzyneczka replied:
I see
I will replace what needs be if this file is unwelcome
didn't know it was a hot topic

PowerfulBacon replied:
As said previously, I'm not as against them as I used to be but at the same time if we are doing this it shouldn't be here.

Szyszkrzyneczka replied:
I'll remove this then

Szyszkrzyneczka replied:
Fixed

Szyszkrzyneczka committed:
Reduces the error count from >1000 down to <680

I am still working on this

Szyszkrzyneczka committed:
300+ erros

new paradigm, closing the gates of hell

Szyszkrzyneczka committed:
100+ errors left

time to port clockcult from yoggers

Szyszkrzyneczka committed:
Part two finished

Now only thing left is bug fixing and merge conflict

Current bugs to fix
Clockwork slab empowerment (spells like kindle etc)
Icons not being correct

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
Catches up to the rest of the codebase

why are you guys so fast

Szyszkrzyneczka committed:
Does stuff to make the CI scream less

What the fuck is a proc helper reference!!!!

Szyszkrzyneczka committed:
CI should scream less

Szyszkrzyneczka committed:
Linters should be fine

Only thing left is to clean up maps

Szyszkrzyneczka committed:
Makes CI less screamy

Szyszkrzyneczka committed:
Almost shuts the CI off

Hopefully I can get to actual bugfixing now

Szyszkrzyneczka committed:
please ci shut uppppp

Szyszkrzyneczka committed:
is there a way to not have to make a singular commit to check if the test will shit itself or not

Szyszkrzyneczka committed:
I hope this is it

please work

Szyszkrzyneczka committed:
fixes mansussy icon not showing up proper

Szyszkrzyneczka committed:
Shoddily makes clockies stupid slab work

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
prodding

Tsar-Salat commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -205,7 +205,8 @@ CREATION_TEST_IGNORE_SELF(/mob/living/carbon)
 
 /mob/living/carbon/on_fall()
 	. = ..()
-	loc.handle_fall(src)//it's loc so it doesn't call the mob's handle_fall which does nothing
```
loc?.handle_fall(src)

Szyszkrzyneczka replied:
I see this first, what does ? exactly do?

Szyszkrzyneczka replied:
Aight I did it that way, thanks for enlightening me

Tsar-Salat commented on code/modules/mob/mob.dm:
```diff
@@ -45,7 +45,8 @@
 	ghostize()
 	if(mind?.current == src) //Let's just be safe yeah? This will occasionally be cleared, but not always. Can't do it with ghostize without changing behavior
 		mind.set_current(null)
-	actions.Cut()
```
actions?.cut()

If you are running into a lot of places where you are just making a 1 line nest, it's cleaner to use use abbrv. ? instead

Szyszkrzyneczka committed:
Mansussy back at it again to ruin my life :sob:

Szyszkrzyneczka committed:
This should be better

PowerfulBacon commented on code/__DEFINES/dcs/signals/signals_atom/signals_atom.dm:
```diff
@@ -9,6 +9,8 @@
+#define COMSIG_ATOM_EXAMINE "atom_examine"
 	//Positions for overrides list
 	#define EXAMINE_POSITION_ARTICLE (1<<0)
 	#define EXAMINE_POSITION_BEFORE (1<<1)
```
The indented defines are actually the return values of signal calls. They should always be below the signal that actually listens to their returns. In this case `EXAMINE_POSITION_ARTICLE` does not belong to `COMSIG_ATOM_EXAMINE`, so it shouldn't be underneath it and it should be under `COMSIG_ATOM_GET_EXAMINE_NAME` since thats the signal that uses it.

Szyszkrzyneczka replied:
Fixed

PowerfulBacon commented on code/__DEFINES/dcs/signals/signals_atom/signals_atom_mouse.dm:
```diff
@@ -0,0 +1,14 @@
+	#define COMPONENT_CANCEL_CLICK_ALT (1<<0)
+///from base of atom/alt_click_secondary(): (/mob)
+#define COMSIG_CLICK_ALT_SECONDARY "alt_click_secondary"
+	#define COMPONENT_CANCEL_CLICK_ALT_SECONDARY (1<<0)
```
Is it worth having these while we don't actually support secondary clicking at all?

Szyszkrzyneczka replied:
Fixed sorry

PowerfulBacon commented on code/__DEFINES/dcs/signals/signals_heretic.dm:
```diff
@@ -0,0 +1,18 @@
+/// Default behavior is to use the hand, so return this to blocks the mansus fist from being consumed after use.
+	#define COMPONENT_BLOCK_HAND_USE (1<<0)
+/// From /obj/item/melee/touch_attack/mansus_fist/afterattack_secondary : (mob/living/source, atom/target)
+#define COMSIG_HERETIC_MANSUS_GRASP_ATTACK_SECONDARY "mansus_grasp_attack_secondary"
```
Here too, since we don't have secondary clicks at all atm

Szyszkrzyneczka replied:
Fixed

PowerfulBacon commented on code/__DEFINES/dcs/signals/signals_mob/signals_human.dm:
```diff
@@ -6,17 +6,13 @@
-	/// Default behavior is to use a charge, so return this to blocks the mansus fist from being consumed after use.
-	#define COMPONENT_BLOCK_CHARGE_USE (1<<0)
+/// Default behavior is to use a charge, so return this to blocks the mansus fist from being consumed after use.
+#define COMPONENT_BLOCK_CHARGE_USE (1<<0)
```
This is a response from a signal and should be indented. It seems like you moved the actual define that this belongs to somewhere else though, so this should be moved with it. Unless you already moved it, in which case just delete this.

PowerfulBacon commented on code/__DEFINES/dcs/signals/signals_mob/signals_living.dm:
```diff
@@ -69,3 +69,25 @@
+///from /obj/structure/door/crush(): (mob/living/crushed, /obj/machinery/door/crushing_door)
+#define COMSIG_LIVING_DOORCRUSHED "living_doorcrush"
+/// Block the electrocute_act() proc from proceeding
+#define COMPONENT_LIVING_BLOCK_SHOCK (1<<0)
```
This should be indented because its a response for shocking. This should also be placed underneath the signal that its actually returned by, which by the name probably isn't DOORCRUSHED.

Szyszkrzyneczka replied:
Fixed

PowerfulBacon commented on code/__DEFINES/hud.dm:
How many of these are actually being used and how much of it is a raw copy and paste from TG?

Szyszkrzyneczka replied:
Fixed

PowerfulBacon commented on code/__DEFINES/traits/sources.dm:
Whats going on with this file? You moved some of them into declarations but not all of them.

Szyszkrzyneczka replied:
I got confused sorry, fixed

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

PowerfulBacon commented on code/__DEFINES/traits/declarations.dm:
Is moving all the trait sources into declarations a good idea, is this what TG does?

Szyszkrzyneczka replied:
I got confused sorry, fixed

PowerfulBacon commented on code/__byond_version_compat.dm:
You shouldn't be modifying this

Szyszkrzyneczka replied:
I forgot to uncheck the file, same with the other one

PowerfulBacon commented on code/_compile_options.dm:
You shouldn't be modifying this

Szyszkrzyneczka replied:
I forgot to uncheck the file, same with the other one

PowerfulBacon commented on code/game/objects/items/mjolnir.dm:
This file is being reverted to worse formatting, why?

Szyszkrzyneczka replied:
Due to tests failing and pointing in the direction of this file

Szyszkrzyneczka replied:
Okie it's gone back to being good

PowerfulBacon commented on code/game/objects/items/storage/holsters.dm:
```diff
@@ -0,0 +1 @@
+///There's probably this file somewhere else in the codebase, treat this like a stickynote
```
The file probably exists somewhere? What's the point of this?

PowerfulBacon commented on code/game/objects/items/storage/bags.dm:
```diff
@@ -259,7 +259,6 @@
 	icon_state = "portaseeder"
-	actions_types = list(/datum/action/item_action/portaseeder_dissolve)
 
 /obj/item/storage/bag/plants/portaseeder/proc/dissolve_contents()
```
So how does this get called now?

Szyszkrzyneczka replied:
Whoopsies

PowerfulBacon commented on code/game/objects/items/tanks/watertank.dm:
```diff
@@ -720,3 +720,40 @@
+	possible_transfer_amounts = list(75,100,150)
+
+/obj/item/watertank/op/make_noz()
+	return new /obj/item/reagent_containers/spray/mister/op(src)
```
This seems like a random feature being added from TG that isn't relevant to the port

Szyszkrzyneczka replied:
sorry didnt notice

PowerfulBacon commented on code/game/objects/structures/crates_lockers/closets.dm:
```diff
@@ -643,3 +643,9 @@
 		dat += "[custom_data ? ",\n[custom_data]" : ""]"
 	return dat
+
+/obj/structure/closet/proc/on_magic_unlock(datum/source, datum/action/cooldown/spell/aoe/knock/spell, mob/living/caster)
```
Where is the signal for this being registered at?

Szyszkrzyneczka replied:
Now it's registered, terribly sorry

PowerfulBacon commented on code/modules/admin/admin_verbs.dm:
```diff
@@ -188,7 +189,7 @@ GLOBAL_PROTECT(admin_verbs_debug)
 	/client/proc/cmd_display_overlay_log,
 	/client/proc/reload_configuration,
-	/client/proc/give_all_spells,
+	///client/proc/give_all_spells,
```
Does this no longer work or is it removed?

Szyszkrzyneczka replied:
Moved to a new panel idk how to make the panel appear by a button press

Szyszkrzyneczka replied:
Sadly I can't seem to give the command to give ALL spells because that freezes my pc :(

PowerfulBacon commented on code/modules/antagonists/clock_cult/items/clockwork_slab.dm:
```diff
@@ -93,6 +93,18 @@ GLOBAL_LIST_INIT(clockwork_slabs, list())
+/obj/item/clockwork/clockwork_slab/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	switch(empowerment)
+		if("kindle")
```
This should use defines rather than random strings

PowerfulBacon commented on code/modules/antagonists/clock_cult/items/clockwork_slab.dm:
```diff
@@ -93,6 +93,18 @@ GLOBAL_LIST_INIT(clockwork_slabs, list())
+		if("compromise")
+			sentinels_compromise(user, target)
+	if(active_scripture)
+		active_scripture.end_invokation()
```
A much better way to do this would be to have some kind of active_scripture.on_attack() and then override the proc to add behaviour for specific spells.

PowerfulBacon commented on code/modules/mob/mob.dm:
```diff
@@ -45,13 +45,7 @@
-		else
-			qdel(A) // Other actions can be safely deleted
-	actions.Cut()
+	actions?.Cut()
```
Is this what TG does? Nothing with the actions?

Szyszkrzyneczka replied:
Apparently they do in fact do nothing with the actions. I don't know if it actually matters though

PowerfulBacon commented on code/modules/mob/mob.dm:
```diff
@@ -785,19 +787,29 @@
+			REF(action.viewers[hud_used]),
+		))
+
+	return data
```
We have completely different stat panels to TG, they are built into the TGUI itself. This is the code you need to convert into using actions. You can't just port what TG has written and expect it to work when you are changing core functionality. You need to know how to update existing code to use your newly added systems, which requires understanding how they work.
```
/**
  * Convert a list of spells into a displyable list for the statpanel
  *
  * Shows charge and other important info
  */
/mob/proc/get_spell_stat_data(list/spells, current_tab)
	var/list/stat_data = list()
	for(var/obj/effect/proc_holder/spell/S in spells)
		if(S.can_be_cast_by(src) && current_tab == S.panel)
			client.stat_update_mode = STAT_MEDIUM_UPDATE
			switch(S.charge_type)
				if("recharge")
					stat_data["[S.name]"] = GENERATE_STAT_TEXT("[S.charge_counter/10.0]/[S.charge_max/10]")
				if("charges")
					stat_data["[S.name]"] = GENERATE_STAT_TEXT("[S.charge_counter]/[S.charge_max]")
				if("holdervar")
					stat_data["[S.name]"] = GENERATE_STAT_TEXT("[S.holder_var_type] [S.holder_var_amount]")
	return stat_data
```

Szyszkrzyneczka replied:
Fixed

PowerfulBacon commented on code/modules/mob/mob_defines.dm:
```diff
@@ -162,14 +163,8 @@ CREATION_TEST_IGNORE_SELF(/mob)
-	var/list/mob_spell_list = list()
-
+	///A weakref to the last mob/living/carbon to push/drag/grab this mob (exclusively used by slimes friend recognition)
+	var/datum/weakref/LAssailant = null
```
This was deleted on purpose, what do you need it for? Just porting a variable declartion won't work when the existing code was removed.

Szyszkrzyneczka replied:
sowwy

PowerfulBacon commented on code/modules/mob/mob_stat.dm:
As I mentioned in a prior review, this is just deleting behaviour and adding the procs from TG but not actually making an attempt to call them. Our stat works differently to TG and you need to know how to update the existing content rather than just blindly copy and pasting what TG does in hopes that it will work when it isn't being referenced at all.

This kind of port really needs:
- Port the core functionality, remove proc_holder
- Manually make the changes that were made on TG across the entire codebase.

Szyszkrzyneczka replied:
Fixed

Szyszkrzyneczka committed:
Whoopsie daisies

Ignore this teehee x3

Szyszkrzyneczka committed:
whuh

PowerfulBacon commented on code/__DEFINES/magic.dm:
```diff
@@ -13,3 +38,60 @@
+#define MAGIC_RESISTANCE (1<<0)
+/// Tinfoil hat magic resistance that blocks mental magic (telepathy / mind links, mind curses, abductors)
+#define MAGIC_RESISTANCE_MIND (1<<1)
+#define MAGIC_RESISTANCE_HOLY (1<<2)
```
You add these but component/anti_magic doesn't actually seem to have been updated? This means that its just not going to work at all because anti_magic expects booleans and not flags.

Szyszkrzyneczka replied:
fixed

PowerfulBacon reviewed (changes requested):
This PR is an extremely messy port. So far I haven't gotten too high hopes that the content of it is actually isolated to what is relevant for the port and I get the feeling that there is going to be a huge amount of content haphazardly added which just isn't used or supported. Theres a lot of inconsistent changes to things like the define, presumably becuase only the things from TG were ported and not everything else or you were working on an outdated branch.

I have only looked through a small part of this PR, similar changes are extremely widespread and likely affect a lot of the content being added. You need to be more careful to differentiate:
- What is necessary to port by just copying from TG (the proc_holder base code)
- What needs manual updating (probably the proc_holders should have been done manually since we are going to have it majorly different to TG)
- What is unnecessary and isn't used by bee.

Where you are porting things you need to be careful that you are actually porting entire PRs. You are porting some kind of anti_magic rework by taking the defines and calling a function with those defines... but not actually porting the bit that makes that work. A large amount of content is going to be extremely broken by this PR in the current state.

Szyszkrzyneczka committed:
Fixes a bunch of balooney that I did

I haven't tested the changes, it's pretty late and I will do that when resolving conversations. Also removes span.dm I am not doing anything with that ever again, what a bunch of doohickey nonsense

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
antimagic finally works back

yippie

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
Action buttons fully work teeheeeee

Szyszkrzyneczka committed:
CI please dont scream :(

Szyszkrzyneczka committed:
Whoopsie doodles forgot about this one

Szyszkrzyneczka committed:
Okay most code stuffs done

Some spells still have icons missing, I will have to go through them

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
makes things a little better

idk what to do now

Szyszkrzyneczka committed:
Removes questionable code

does anyone know where I can get missing sprites from....

Szyszkrzyneczka committed:
Fixes icons

Everything should be dandy

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
All done and set

Should be good and dandy

Szyszkrzyneczka committed:
Fixes the wizard issue

Szyszkrzyneczka committed:
Merge branch 'master' into Future

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
reverts shock touch because i didnt notice whoopsies

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
bleeeh (merge this pr I am tired of having to readd the stupid fucking books each time a map gets updated)

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
Update EchoStation.dmm

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
Smhsmhsmhsmhsmh Tell me when you chance clockies

Tsar-Salat commented on code/__DEFINES/actions.dm:
```diff
@@ -1,8 +1,50 @@
+
+
+///Action button triggered with right click
+#define TRIGGER_SECONDARY_ACTION (1<<0)
```
Does this currently work with our context menu? I assume this is for trigger flags when right clicking action buttons, but does it do anything?

If not, I recommend commenting it out for now

Szyszkrzyneczka replied:
I forgot to remove this define when removing all mentions of right clicking

Tsar-Salat commented on code/__DEFINES/combat.dm:
```diff
@@ -284,8 +284,21 @@ GLOBAL_LIST_INIT(shove_disarming_types, typecacheof(list(
+/// Alternate attack defines. Return these at the end of procs like afterattack_secondary.
+/// Calls the normal attack proc. For example, if returned in afterattack_secondary, will call afterattack.
+/// Will continue the chain depending on the return value of the non-alternate proc, like with normal attacks.
+#define SECONDARY_ATTACK_CALL_NORMAL 1
```
Ditto

Szyszkrzyneczka replied:
ditto

Tsar-Salat commented on code/__DEFINES/flags.dm:
```diff
@@ -99,19 +99,20 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
 /// If megafauna can be spawned by natural random generation
 #define MEGAFAUNA_SPAWN_ALLOWED		(1<<5)
+/// Are you forbidden from teleporting to the area? (centcom, mobs, wizard, hand teleporter)
+#define NOTELEPORT (1<<6)
```
![Screenshot 2024-11-10 152653](https://github.com/user-attachments/assets/ff1a7aab-0e55-4e39-99c2-bd4dcd0c2189)

![Screenshot 2024-11-10 152701](https://github.com/user-attachments/assets/27ad8851-241e-44f1-984b-809a3db38057)

This flag may be unecessary as we already have a tiered var on areas for checking if a mob can teleport there

Szyszkrzyneczka replied:
You are absolutely right bestie

Tsar-Salat commented on code/__DEFINES/obj_flags.dm:
```diff
@@ -50,6 +50,8 @@
 #define MASKEXTENDRANGE			(1<<13) //! For masks, allows you to breathe from internals on adjecent tiles
 #define NOTCONSUMABLE			(1<<14) //! Moths cannot eat clothing with that flag
+/// Usable as casting clothes by wizards (matters for suits, glasses and headwear)
+#define CASTING_CLOTHES (1<<15)
```
Credit the author

https://github.com/tgstation/tgstation/pull/66604

Szyszkrzyneczka replied:
Didn't even know this was a pr
Nevertheless I will, thanks for the heads up

Tsar-Salat reviewed (changes requested):
I found a few minor things. Otherwise this looks ready for TM

Szyszkrzyneczka committed:
Requested changes

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
Everything is nice and dandy :)

Sarchutar commented on tgui/packages/tgui/interfaces/Spellbook.js:
This whole file is behind on whats currently on the server, see #11716

Sarchutar commented on code/modules/antagonists/wizard/equipment/wizard_spellbook.dm:
```diff
@@ -0,0 +1,329 @@
+		individual_entry_data["cooldown"] = entry.cooldown
+		individual_entry_data["cat"] = entry.category
+		individual_entry_data["refundable"] = entry.refundable
+		individual_entry_data["buyword"] = entry.buy_word
```
```suggestion
		individual_entry_data["refundable"] = entry.refundable
		individual_entry_data["limit"] = entry.limit
		individual_entry_data["buyword"] = entry.buy_word
```
limit needs to be sent as well if the updated version of the tgui file is pulled

Sarchutar commented on code/modules/antagonists/wizard/equipment/wizard_spellbook.dm:
```diff
@@ -0,0 +1,329 @@
+
+	/// The number of book charges we have to buy spells
+	var/uses = 10
+	/// The bonus that you get from going semi-random.
+	var/semi_random_bonus = 2
+	/// The bonus that you get from going full random.
+	var/full_random_bonus = 5
```
These two got completely removed in the spellbook tgui pr because wild magic was still a thing there, since wild magic gets removed in this pr it would also need to be added back to the new tgui file

Szyszkrzyneczka replied:
Are you sure the random thing got removed
  {
    title: 'Randomize',
    blurb: "If you didn't like the loadouts offered, you can embrace chaos. Not recommended for newer wizards.",
    component: () => <Randomize />,
  },

from your .js

I can remove those if you want, But doesn't seem like removed to me

Sarchutar commented on code/modules/antagonists/wizard/equipment/wizard_spellbook.dm:
```diff
@@ -0,0 +1,329 @@
+		individual_entry_data["name"] = entry.name
+		individual_entry_data["desc"] = entry.desc
+		individual_entry_data["ref"] = REF(entry)
+		individual_entry_data["requires_wizard_garb"] = entry.requires_wizard_garb
```
`requires_wizard_garb` is `clothes_req` in the newer tgui file so would also need to be changed there as well

Sarchutar reviewed (commented):
Some spellbook tgui stuff that is behind from what is on the server

Szyszkrzyneczka committed:
Resolves conflicts

PowerfulBacon reviewed (changes requested):
This PR breaks anti-magic overlays. When you pickup a bible, you should be able to see your anti-magic aura but that is now gone:

![image](https://github.com/user-attachments/assets/623340c1-9bdd-4f10-921a-4051e07b2685)

PowerfulBacon reviewed (changes requested):
This PR also erases a change to targetted spells I did ages ago that made them use click targetting rather than select from a dropdown list.

Replication:
- Spawn as wizard
- Choose the cluwne curse
- Try to cast it
- On the server you get a crosshair, on this new branch you get a pop-up list instead.

All targetted spells shold use crosshair targetting rather than giving the user a list that forces them to pause. How you implement this doesn't matter, but the list selection spell type should be completely removed in favour of forcing spell coders to use crosshairs.

PowerfulBacon reviewed (changes requested):
This PR seems to remove netcaster spiders from spider eggs

PowerfulBacon reviewed (changes requested):
Considering you removed the debug spell book which exists for testing spells, that doesn't raise my hope in how well-tested this PR is

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
Brings this pr up to speed

Szyszkrzyneczka committed:
slight fixes

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Tsar-Salat commented on code/game/objects/items.dm:
```diff
@@ -680,10 +724,11 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
+	for(var/datum/action/action_item_has as anything in actions)
+		action_item_has.Remove(user)
+	if(item_flags & DROPDEL)
+		qdel(src)
```
```suggestion
		action_item_has.Remove(user)
```

Tsar-Salat replied:
This is a duplicate, we already handle DROPDEL at the bottom of the proc.

This is whats causing signal runtimes and your qdel loop

Szyszkrzyneczka replied:
Fuck

Szyszkrzyneczka committed:
fuck

Szyszkrzyneczka committed:
Uhhhhhhh

Szyszkrzyneczka committed:
Fixes mouse pointers

Szyszkrzyneczka committed:
Update mob.dm

PowerfulBacon commented on code/modules/spells/spell_types/bee_spells/cluwnecurse.dm:
```diff
@@ -1,29 +1,20 @@
-	cast_on.cluwneify()
+	if(cast_on == owner)
+		if(tgui_alert(usr, "Are you sure you want to curse yourself to become a cluwne?", "Fool's choice", list("Yes!", "No...")) == "Yes")
+			cast_on.cluwneify()
```
This breaks the spell if you use it on someone other than yourself

Szyszkrzyneczka replied:
so that's why it failed to cluwne people

Szyszkrzyneczka committed:
tiny fixes, should cover all, vid soon

Tsar-Salat commented on code/_onclick/hud/_defines.dm:
Why did 80% of the defines move and the remaining get left behind?

If you are moving the defines, don't leave *just* the bee-specific ones. I think you followed TG a little too closely here.

Tsar-Salat commented on code/_onclick/hud/action_button.dm:
```diff
@@ -1,174 +1,145 @@
 	usr.next_click = world.time + 1
+
 	linked_action.Trigger()
-	SEND_SOUND(usr, 'sound/effects/pop.ogg')
```
where did this end up? 
- https://github.com/BeeStation/BeeStation-Hornet/pull/2377

If we don't send a noise anymore onclick thats fine, but if it got moved, make sure it still uses this sound.

Szyszkrzyneczka replied:
Ughhhhh  I forgot

Tsar-Salat commented on code/_onclick/hud/action_button.dm:
```diff
@@ -195,60 +166,241 @@
+			for(var/datum/hud/hud as anything in action.viewers)
+				var/atom/movable/screen/movable/action_button/button = action.viewers[hud]
+				hud.position_action(button, SCRN_OBJ_DEFAULT)
+		to_chat(usr, ("<span class='notice'>Action button positions have been reset.</span>"))
```
```suggestion
		to_chat(usr, "<span class='notice'>Action button positions have been reset.</span>")
```

Szyszkrzyneczka committed:
minor fixes i noticed

Tsar-Salat reviewed (changes requested):
A few minor comments, with one major one.

![Screenshot 2024-12-06 170153](https://github.com/user-attachments/assets/6950259c-b61a-4280-ae2a-7eda2cb69716)


The maptext for the timer is rendering *behind* the icon foreground.


This obscures the timer, with some you can read, with others you cannot even make out the timer. This should be fixed, and if possible I would choose a more legible color for the text

Szyszkrzyneczka committed:
Restores the things i accidentally deleted

PowerfulBacon committed:
08/12/2024

PowerfulBacon committed:
08/12/2024-2

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

PowerfulBacon committed:
Removes action/cooldown

Szyszkrzyneczka committed:
Fixes various things I found (mainly smoke!!!)

PowerfulBacon committed:
Some more refactor work

PowerfulBacon committed:
Final refactors

PowerfulBacon committed:
Merge branch 'Future' into cooldown-action-refactor

Szyszkrzyneczka committed:
Merge pull request #1 from PowerfulBacon/cooldown-action-refactor

Cooldown action refactor

Szyszkrzyneczka committed:
Update debug.dm

PowerfulBacon committed:
Fixes compile errors

PowerfulBacon committed:
Fixes various issues related to incorrect targets being selected

PowerfulBacon committed:
Fixes various issues related to incorrect targets being selected

PowerfulBacon committed:
Merge branch 'Future' of https://github.com/Szyszkrzyneczka/BeeStation-Hornet into pr/11527

PowerfulBacon committed:
Fixes compilation errors

PowerfulBacon committed:
Adds a give all mutations button for mutation testing

PowerfulBacon committed:
Fixes spells

PowerfulBacon committed:
Fixes strained muscles and cooldown text styling

PowerfulBacon committed:
Fixes strained muscles and augmented eyesight

Szyszkrzyneczka committed:
updates anti-magic component properly this time

PowerfulBacon committed:
Updates the debug verb to only grant action mutations

PowerfulBacon committed:
Fixes the positioning of blood magic

PowerfulBacon committed:
Fixes the stat panel

Szyszkrzyneczka committed:
minor things

PowerfulBacon committed:
Fixes augry item action

PowerfulBacon committed:
Merge branch 'Future' of https://github.com/Szyszkrzyneczka/BeeStation-Hornet into pr/11527

Szyszkrzyneczka committed:
separates mutation verbs

still no fix on teleport+wiz battlesuit interaction

PowerfulBacon committed:
Fixes antimagic issues on wizards

PowerfulBacon committed:
Fixes ghosting during spellcasts

Szyszkrzyneczka committed:
fixes the missing icon

Szyszkrzyneczka committed:
Fixes clockie runtimes and bugs

Szyszkrzyneczka committed:
Fixes weldhat issue and #12054

Szyszkrzyneczka committed:
closes #12055

Szyszkrzyneczka committed:
fixes wronged touch spells that I know of

Tsar-Salat reviewed (dismissed):
![image](https://github.com/user-attachments/assets/680479ce-a6b7-426f-b9e6-4ff1520d215a)

This PR seems to arbitrarily reorder and delete some of the ghost orbit menu UI buttons. Likely an accidental port. 

I would revert please.

Szyszkrzyneczka committed:
Fixes the UI whoopsy

Szyszkrzyneczka committed:
Update hud.dm

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
Merge remote-tracking branch 'upstream/master' into Future

Szyszkrzyneczka committed:
Fixes stuff and hopefully doesnt break anything

Szyszkrzyneczka committed:
I wonder how much will be inevitably lost when this gets merged

Szyszkrzyneczka committed:
whoopsie

Szyszkrzyneczka committed:
Merge branch 'master' into Future

Szyszkrzyneczka committed:
Adds spyware

Szyszkrzyneczka committed:
Revert "Adds spyware"

This reverts commit 783ff89ec958d08b049402b1f3204cd143e31a2c.

Szyszkrzyneczka committed:
removes spyware

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into Future

PowerfulBacon committed:
Fixes some spells modifying the background of the spell instead of the foreground

PowerfulBacon committed:
Lesser lichdom can be cast while you have anti-magic items on you.

PowerfulBacon committed:
Merge remote-tracking branch 'upstream/master' into pr/11527

PowerfulBacon committed:
Some fixes

Tsar-Salat committed:
rkz fixes <3

PowerfulBacon committed:
Re-executes update paths

PowerfulBacon committed:
Update FlandStation.dmm

PowerfulBacon committed:
Merge remote-tracking branch 'upstream/master' into pr/11527

PowerfulBacon committed:
Merge remote-tracking branch 'upstream/master' into pr/11527

PowerfulBacon committed:
Fixes xenoarch not updating its buttons

PowerfulBacon reviewed (approved):
Been testmerged a few times and all outstanding issues that I knew of were resolved. Any merge skews would be picked up by the fact that most action procs were renamed to follow DM formatting standards

PowerfulBacon committed:
Merge remote-tracking branch 'upstream/master' into pr/11527

PowerfulBacon committed:
Adds the art of thievery back

PowerfulBacon committed:
Update thievery.dm

PowerfulBacon committed:
Resets maps

Rukofamicom commented on code/modules/mob/living/carbon/human/species_types/lizardpeople.dm:
```diff
@@ -20,7 +20,7 @@
 	meat = /obj/item/food/meat/slab/human/mutant/lizard
 	skinned_type = /obj/item/stack/sheet/animalhide/lizard
 	exotic_bloodtype = "L"
-	inert_mutation = FIREBREATH
```
Why was this changed and not documented?

Rukofamicom commented on code/modules/mob/living/carbon/human/species_types/oozelings.dm:
```diff
@@ -19,7 +19,7 @@
 	changesource_flags = MIRROR_BADMIN | WABBAJACK | MIRROR_PRIDE | MIRROR_MAGIC | RACE_SWAP | ERT_SPAWN | SLIME_EXTRACT
 	species_language_holder = /datum/language_holder/oozeling
 	swimming_component = /datum/component/swimming/dissolve
-	inert_mutation = ACIDOOZE
```
And this too?

Rukofamicom reviewed (commented):
Undocumented and totally unrelated changes are not cool yo.

DeltaFire15 commented on code/modules/antagonists/clock_cult/scriptures/kindle.dm:
```diff
@@ -1,71 +0,0 @@
-	cogs_required = 1
-	category = SPELLTYPE_SERVITUDE
-
-/datum/clockcult/scripture/slab/kindle/apply_effects(atom/A)
```
kindle: prior effects

DeltaFire15 commented on code/modules/antagonists/clock_cult/scriptures/slab_empowerments/kindle.dm:
```diff
@@ -0,0 +1,45 @@
+
+//For the Kindle scripture; stuns and mutes a clicked_on non-servant.
+
+/datum/clockcult/scripture/slab/proc/kindle(mob/living/caller, mob/living/clicked_on)
+	empowerment = null
+	to_chat(caller, ("<span class='brass'>You release the light of Ratvar!</span>"))
+	clockwork_say(caller, text2ratvar("Purge all untruths and honor Engine!"))
+	if(isliving(clicked_on))
+		var/mob/living/L = clicked_on
+		if(is_servant_of_ratvar(L) || L.stat)
+			return BULLET_ACT_HIT
+		var/atom/O = L.can_block_magic(MAGIC_RESISTANCE_HOLY)
+		playsound(L, 'sound/magic/fireball.ogg', 50, TRUE, frequency = 1.25)
+		if(O)
+			if(isitem(O))
+				L.visible_message(("<span class='warning'>[L]'s eyes flare with dim light!</span>"), \
+				("<span class='userdanger'>Your [O] glows white-hot against you as it absorbs [src]'s power!</span>"))
+			else if(ismob(O))
+				L.visible_message(("<span class='warning'>[L]'s eyes flare with dim light!</span>"))
+			playsound(L, 'sound/weapons/sear.ogg', 50, TRUE)
+		else
+			L.visible_message(("<span class='warning'>[L]'s eyes blaze with brilliant light!</span>"), \
+			("<span class='userdanger'>Your vision suddenly screams with white-hot light!</span>"))
+			L.Paralyze(5 SECONDS)
+			L.flash_act(1, 1)
+			if(iscultist(L))
+				L.adjustFireLoss(15)
+	return TRUE
```
new effects. Undocumented changes I can tell:

- No longer mutes despite stating so in description.
- Paralyze time reduced to `5 SECONDS` instead of 150 (aka `15 SECONDS`).
- Now stuns blood cultists and damages them, vs. previous debilitation with no stun.
- No longer exposes blood cult allegiance (besides quiet fire damage).
- No longer causes reduced effects on reebe.
- No longer causes special effects to silicons (EMP).
- Can no longer be used on incapacitated targets (stat > 0).
- No longer resisted by mindshielded targets.

DeltaFire15 commented on code/modules/antagonists/clock_cult/scriptures/sentinels_compromise.dm:
```diff
@@ -1,54 +0,0 @@
-				clockwork_say(invoker, text2ratvar(after_use_text), TRUE)
-			end_invokation()
-
-/datum/clockcult/scripture/slab/sentinelscompromise/apply_effects(mob/living/M)
```
sent compromise prior effects

DeltaFire15 commented on code/modules/antagonists/clock_cult/scriptures/slab_empowerments/sentinels_compromise.dm:
```diff
@@ -0,0 +1,63 @@
+		if(!is_servant_of_ratvar(L))
+			to_chat(caller, ("<span class='inathneq'>\"[L] does not yet serve Ratvar.\"</span>"))
+			return TRUE
+		if(L.stat == DEAD)
+			to_chat(caller, ("<span class='inathneq'>\"[L.p_theyre(TRUE)] dead. [text2ratvar("Oh, child. To have your life cut short...")]\"</span>"))
+			return TRUE
+
+		var/brutedamage = L.getBruteLoss()
+		var/burndamage = L.getFireLoss()
+		var/oxydamage = L.getOxyLoss()
+		var/totaldamage = brutedamage + burndamage + oxydamage
+		if(!totaldamage && (!L.reagents || !L.reagents.has_reagent(/datum/reagent/water/holywater)))
+			to_chat(caller, ("<span class='inathneq'>\"[L] is unhurt and untainted.\"</span>"))
+			return TRUE
+		to_chat(caller, ("<span class='brass'>You bathe [L == caller ? "yourself":"[L]"] in Inath-neq's power!</span>"))
+		var/clicked_onturf = get_turf(L)
+		var/has_holy_water = (L.reagents && L.reagents.has_reagent(/datum/reagent/water/holywater))
+		var/healseverity = max(round(totaldamage*0.05, 1), 1) //shows the general severity of the damage you just healed, 1 glow per 20
+		for(var/i in 1 to healseverity)
+			new /obj/effect/temp_visual/heal(clicked_onturf, "#1E8CE1")
+		if(totaldamage)
+			L.adjustBruteLoss(-brutedamage, TRUE, FALSE)
+			L.adjustFireLoss(-burndamage, TRUE, FALSE)
+			L.adjustOxyLoss(-oxydamage)
+			caller.adjustToxLoss(totaldamage * 0.5, TRUE, TRUE)
+			clockwork_say(caller, text2ratvar("[has_holy_water ? "Heal tainted" : "Mend wounded"] flesh!"))
+			log_combat(caller, L, "healed with Sentinel's Compromise")
+			L.visible_message(("<span class='warning'>A blue light washes over [L], [has_holy_water ? "causing [L.p_them()] to briefly glow as it mends" : " mending"] [L.p_their()] bruises and burns!</span>"), \
+			("<span class='heavy_brass'>You feel Inath-neq's power healing your wounds[has_holy_water ? " and purging the darkness within you" : ""], but a deep nausea overcomes you!</span>"))
+		else
+			clockwork_say(caller, text2ratvar("Purge foul darkness!"))
+			log_combat(caller, L, "purged of holy water with Sentinel's Compromise")
+			L.visible_message(("<span class='warning'>A blue light washes over [L], causing [L.p_them()] to briefly glow!</span>"), \
+			("<span class='heavy_brass'>You feel Inath-neq's power purging the darkness within you!</span>"))
+		playsound(clicked_onturf, 'sound/magic/staff_healing.ogg', 50, 1)
+
+		if(has_holy_water)
+			L.reagents.remove_reagent(/datum/reagent/water/holywater, 1000)
+
+	return TRUE
```
New effects. Undocumented changes as far as I can tell:

- No longer affects cloneloss
- Now purges ALL brute / fire / oxyloss as opposed to 60% each.
- No longer has a damage cap for the toxdamage from the conversion (previously 80)
- No longer negates several non-pure-damage effects, such as: bloodloss, low food, blind / blur / dizzy, husking, hallucinations, nearsightedness and bodytemperature difference.

DeltaFire15 reviewed (commented):
This PR seems to have significantly altered the effects of multiple clockwork abilities with no mention thereof in the PR body. Was noticing some strange discrepancies between a version I was testing stuff on and beecode, so I took a look and traced it to this PR.

Elaborating some in review comments, but I wouldn't doubt there being more cases of this (without even mentioning all the changes to phrasing), I've just had enough codediving for a day.


Thats a yikes from me.
